python_type_check CI is broken

This looks like an upstream change that broke CI, see https://app.circleci.com/pipelines/github/pytorch/vision/4305/workflows/9ae30f70-6fa0-419f-966b-8db2e249a12b/jobs/254769 for an example

The error we get is
```
torchvision/datasets/mnist.py:446: error:
Incompatible
return
value
type
(got
"GzipFile",
expected
"IO[Any]")

[return-value]
            return gzip.open(path, 'rb')
                   ^
Found 1 error in 1 file (checked 97 source files)
```

cc @seemethere @pmeier

--- a/torchvision/datasets/mnist.py
+++ b/torchvision/datasets/mnist.py
@@ -7,6 +7,8 @@ import numpy as np
 import torch
 import codecs
 import string
+import gzip
+import lzma
 from typing import Any, Callable, Dict, IO, List, Optional, Tuple, Union
 from .utils import download_url, download_and_extract_archive, extract_archive, \
     verify_str_arg
@@ -435,17 +437,15 @@ def get_int(b: bytes) -> int:
     return int(codecs.encode(b, 'hex'), 16)
 
 
-def open_maybe_compressed_file(path: Union[str, IO]) -> IO:
+def open_maybe_compressed_file(path: Union[str, IO]) -> Union[IO, gzip.GzipFile]:
     """Return a file object that possibly decompresses 'path' on the fly.
        Decompression occurs when argument `path` is a string and ends with '.gz' or '.xz'.
     """
     if not isinstance(path, torch._six.string_classes):
         return path
     if path.endswith('.gz'):
-        import gzip
         return gzip.open(path, 'rb')
     if path.endswith('.xz'):
-        import lzma
         return lzma.open(path, 'rb')
     return open(path, 'rb')
 

